state-chain/pallets/cf-swapping/src/tests/fees.rs

```diff
@@ -484,70 +488,164 @@ fn test_network_fee_calculation() {
 }
 
 #[test]
-fn test_calculate_input_for_desired_output() {
+// Test Swap simulation,
```

Comment:
We usually put comments above directives like `#[test]`. Also should probably use doc comments here.